ai-24sp/assignments/whereismyprozac/week9/gpt_train.py

```diff
@@ -221,35 +219,26 @@ def main(gpt_config, settings):
     ###########################
     # Initiate training
     ###########################
-
-    #train_losses, val_losses, tokens_seen, model = main(GPT_CONFIG_124M, OTHER_SETTINGS)
-    device = None
-    tokenizer = None
-
-    ###########################
-    # After training
-    ###########################
     print(f"argv {sys.argv}")
     if (sys.argv[-1] == "train"):
+        print("Training ")
         train_losses, val_losses, tokens_seen, model, device, tokenizer = main(GPT_CONFIG_124M, OTHER_SETTINGS)
-    ###########################
-    # After training
-    ###########################
 
-    # Plot results
-    epochs_tensor = torch.linspace(0, OTHER_SETTINGS["num_epochs"], len(train_losses))
-    plot_losses(epochs_tensor, tokens_seen, train_losses, val_losses)
-    plt.savefig("loss.pdf")
+        ###########################
+        # After training
+        ###########################
 
-    # Save and load model
-    torch.save(model.state_dict(), "model.pth")
-else:
-    device, tokenizer = get_standard_model_params()
+        # Plot results
+        epochs_tensor = torch.linspace(0, OTHER_SETTINGS["num_epochs"], len(train_losses))
+        plot_losses(epochs_tensor, tokens_seen, train_losses, val_losses)
+        plt.savefig("loss.pdf")
 
-    model = GPTModel(GPT_CONFIG_124M)
+        # Save and load model
+        torch.save(model.state_dict(), "model.pth")
+    else:
+        (device, tokenizer) = get_standard_model_params()
 
     model.load_state_dict(torch.load("model.pth"))
```

Comment:
This is where you can load a model and run it, instead of training it every time.

You can run your python script with an extra command-line arg "train" to do the expensive training, and if you leave it off, it just runs inference.

```
python3 gpt_train.py train
```

vs.

```
python3 gpt_train.py train
```